### PR TITLE
Add tests and rules for downloads, current team and push tokens sections

### PR DESCRIPTION
**Pull request by maaz-bin-asad**
- Added a function to validate fields of downloads page.
- Added create, read and update rules and tests for downloads, current team and push tokens section.
- Changed the validation of 'platform' field from bool to string.

### CODE REVIEW

File: firestore.rules [hunk 1]
```diff
@@ -88,6 +88,12 @@ service cloud.firestore {
       docData.regFormLink is string &&
       docData.startTime is string;
     }
+    function downloadFieldsTypeValidate(docData) {
```
Comment (iamareebjamal) [inline firestore.rules:91]: No space above function

File: firestore.rules [hunk 2]
```diff
@@ -121,7 +126,8 @@ service cloud.firestore {
 
     match /pushTokens/{pushTokensId} {
       allow read: if isAdmin();
-      allow create, update: if documentFieldsCheckOut(["androidId", "createdAt", "deviceToken", "platform"], []) && pushTokenFieldsTypeValidate(request.resource.data);
+      allow create: if isAdmin() && documentFieldsCheckOut(["androidId","createdAt","deviceToken","platform"],[]) && pushTokenFieldsTypeValidate(request.resource.data);
```
Comment (iamareebjamal) [inline firestore.rules:129]: In english, we add a space after comma. But in programming, we forget this rule. Why?

File: firestore.rules [hunk 3]
```diff
@@ -121,7 +126,8 @@ service cloud.firestore {
...
+      allow update: if isAdmin() && editOnlyChangesField(["androidId","createdAt","deviceToken","platform"]) && request.resource.data.androidId==resource.data.androidId && pushTokenFieldsTypeValidate(request.resource.data);
```
Comment (iamareebjamal) [inline firestore.rules:130]: space around ==

File: firestore.rules [hunk 4]
```diff
@@ -149,8 +155,9 @@ service cloud.firestore {
     }
 
     match /downloads/{downloadsId} {
-      allow read;
-      allow create, update: if isAdmin();
+    	allow read;
+    	allow create: if isAdmin() && documentFieldsCheckOut(["file", "name", "size", "url"],[]) && downloadFieldsTypeValidate(request.resource.data);
```
Comment (iamareebjamal) [inline firestore.rules:159]: Why mixing tabs and spaces? I don't understand

Comment (iamareebjamal) [inline firestore.rules:159]: Here, we have space after comma, but not before `[`. Why the inconsistency?

File: tests/test.js [hunk 5]
```diff
@@ -516,7 +516,210 @@ describe('AMURoboclub app DB Unit Testing', () => {
...
+    const admin = getAdminFirestore();
+
+    var current_timestamp = new Date();
+    const testRead = db.collection('/pushTokens').doc(pushTokensId);
+    await admin.collection('/pushTokens').doc(pushTokensId).set({ androidId : "id", createdAt : current_timestamp, deviceToken : "token", platform : "android" });
+    await firebase.assertFails(testRead.update({  location : "location" }));
+  });
+
+
+  });
```
Comment (iamareebjamal) [inline tests/test.js:722]: Not indented correctly I believe

File: firestore.rules (hunk 2 above)
Comment (iamareebjamal) [reply firestore.rules:129]: In fact, here you have removed the spaces after comma.

**Comment by maaz-bin-asad** [comment]
I have updated my PR with required changes. Please have a look.

File: firestore.rules [hunk 6]
```diff
@@ -121,7 +127,8 @@ service cloud.firestore {
 
     match /pushTokens/{pushTokensId} {
       allow read: if isAdmin();
-      allow create, update: if documentFieldsCheckOut(["androidId", "createdAt", "deviceToken", "platform"], []) && pushTokenFieldsTypeValidate(request.resource.data);
+      allow create: if isAdmin() && documentFieldsCheckOut([ "androidId", "createdAt", "deviceToken", "platform" ],[]) && pushTokenFieldsTypeValidate(request.resource.data);
+      allow update: if isAdmin() && editOnlyChangesField([ "androidId", "createdAt", "deviceToken", "platform" ]) && request.resource.data.androidId == resource.data.androidId && pushTokenFieldsTypeValidate(request.resource.data);
```
Comment (iamareebjamal) [inline firestore.rules:131]:
```suggestion
      allow update: if isAdmin() && editOnlyChangesField(["androidId", "createdAt", "deviceToken", "platform"]) && request.resource.data.androidId == resource.data.androidId && pushTokenFieldsTypeValidate(request.resource.data);
```

File: firestore.rules [hunk 7]
```diff
@@ -150,7 +157,8 @@ service cloud.firestore {
 
     match /downloads/{downloadsId} {
       allow read;
-      allow create, update: if isAdmin();
+    	allow create: if isAdmin() && documentFieldsCheckOut([ "file", "name", "size", "url" ],[]) && downloadFieldsTypeValidate(request.resource.data);
+      allow update: if isAdmin() && editOnlyChangesField([ "file", "name", "size", "url" ]) && downloadFieldsTypeValidate(request.resource.data);
```
Comment (iamareebjamal) [inline firestore.rules:161]:
```suggestion
      allow update: if isAdmin() && editOnlyChangesField(["file", "name", "size", "url"]) && downloadFieldsTypeValidate(request.resource.data);
```

**Comment by maaz-bin-asad** [comment]
Updated the suggested changes.

**Pull request merged**